### DISCUSSION TITLE
Upgrading to Phalcon 4 Issue Call to undefined method Phalcon\Cache::hasattribute()

### DISCUSSION THREAD

**Discussion by Nano-GG** [phalcon/cphalcon#15325] [Q&A]
Hello,

So, i'm in the process of migration my personal website from Phalcon 2 with php 5 to Phalcon4 and php7.2, my site was working perfectly but.... well, i think i delayed this to many times.... 

So, leaving apart that upgrading a huge site like mine is a really annoying process, at this moment i have 2 days full time trying to and doing my best, im stuck with this error:

`Fatal error: Uncaught RuntimeException: Call to undefined method Phalcon\Cache::hasattribute() in /home/simpleweb/public_html/app/controllers/IndexController.php:31 Stack trace: #0 [internal function]: Phalcon\Mvc\Model\Query->_getQualified(Array) #1 [internal function]: Phalcon\Mvc\Model\Query->_getExpression(Array) #2 [internal function]: Phalcon\Mvc\Model\Query->_getOrderClause(Array) #3 [internal function]: Phalcon\Mvc\Model\Query->_prepareSelect() #4 [internal function]: Phalcon\Mvc\Model\Query->parse() #5 [internal function]: Phalcon\Mvc\Model\Query->execute() #6 /home/simpleweb/public_html/app/controllers/IndexController.php(31): Phalcon\Mvc\Model::find(Array) #7 [internal function]: IndexController->indexAction() #8 [internal function]: Phalcon\Dispatcher\AbstractDispatcher->callActionMethod(Object(IndexController), 'indexAction', Array) #9 [internal function]: Phalcon\Dispatcher\AbstractDispatcher->dispatch() #10 /home/simpleweb/public_html/public/index.php(17): Phalcon\M in /home/simpleweb/public_html/app/controllers/IndexController.php on line 31`

**IndexController.php:31**

			$news = News::find([
                    'order' => 'id DESC',
			'limit' => 10,
			//'cache' => array('lifetime' => $key['ttl'], 'key' => $key['key1'])
		]);`

I found weird this call Phalcon\Cache::hasattribute(), since i check the docs and hasattribute method is from Phalcon\Mvc\Model\ModelInterface.

Thank you very much for the time of anyone who will try to give me some idea on how to fix this.

**Comment by Jeckerson** [comment]
Hello, check this upgrade guide, might help

https://docs.phalcon.io/4.0/en/upgrade

**Comment by Nano-GG** [comment]
```

<br />
<b>Fatal error</b>:  Uncaught RuntimeException: Call to undefined method Phalcon\Cache::hasattribute() in /home/simpleweb/public_html/app/controllers/IndexController.php:34
Stack trace:
#0 [internal function]: Phalcon\Mvc\Model\Query-&gt;_getQualified(Array)
#1 [internal function]: Phalcon\Mvc\Model\Query-&gt;_getExpression(Array)
#2 [internal function]: Phalcon\Mvc\Model\Query-&gt;_getOrderClause(Array)
#3 [internal function]: Phalcon\Mvc\Model\Query-&gt;_prepareSelect()
#4 [internal function]: Phalcon\Mvc\Model\Query-&gt;parse()
#5 [internal function]: Phalcon\Mvc\Model\Query-&gt;execute()
#6 /home/simpleweb/public_html/app/controllers/IndexController.php(34): Phalcon\Mvc\Model::find(Array)
#7 [internal function]: IndexController-&gt;indexAction()
#8 [internal function]: Phalcon\Dispatcher\AbstractDispatcher-&gt;callActionMethod(Object(IndexController), 'indexAction', Array)
#9 [internal function]: Phalcon\Dispatcher\AbstractDispatcher-&gt;dispatch()
#10 /home/simpleweb/public_html/public/index.php(17): Phalcon\M in <b>/home/simpleweb/public_html/app/controllers/IndexController.php</b> on line <b>34</b><br />
```

**Comment by Nano-GG** [comment]
I used https://github.com/diplopito/Phalcon-Upgrade-Adviser and check all that the report log told me, thank you.

My problem persist anyway, i remove all view cache related, but my problem persist, this is my services and cache files:

- Services: https://pastebin.com/hMAC92XR
- Cache: https://pastebin.com/mn9pnNst

**Comment by quasipickle** [comment]
Comment out all of `cache.php`.  The error is being reported as happening on line 34, but I bet it's actually being caused somewhere else - your custom Cache services might be a good place to start.
